### DISCUSSION TITLE
virtual keyboard removed from Godot 4 [SOLVED]

### DISCUSSION THREAD

**Discussion by J1eC** [godotengine/godot-proposals#5885] [GUI]
Can't use the search functionality of a popup menu on Android without OS.show_virtual_keyboard which was in 3.5 but for some reason was dropped from Godot 4. One Possible workaround would be to add a keyboard popup method to the popup menu node, though if there's no reason the virtual keyboard was removed I would like to request OS.show_virtual_keyboard be added to Godot 4.

**Comment by Calinou** [comment]
This still exists in Godot 4 as `DisplayServer.virtual_keyboard_show()`. This is how LineEdit displays the virtual keyboard internally: https://github.com/godotengine/godot/blob/1e3919c409fa1f24db8ecde58f723fd59adfeccb/scene/gui/line_edit.cpp#L1585-L1593

**Reply by J1eC** [reply]
thanks! 
I haven't done much with servers in the past and couldn't find anything on the docs
